packages/sdk_connectivity_cubit/lib/src/sync_manager.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+import 'package:flutter_fgbg/flutter_fgbg.dart';
+
+const syncIntervalSeconds = 60;
+
+class SyncManager {
+  StreamSubscription<FGBGType>? _lifecycleSubscription;
+  StreamSubscription<List<ConnectivityResult>>? _networkSubscription;
+  DateTime _lastSync = DateTime.fromMillisecondsSinceEpoch(0);
+
+  final BindingLiquidSdk? wallet;
+
+  SyncManager(this.wallet);
+
+  void startSyncing() {
+    _lifecycleSubscription = FGBGEvents.stream.skip(1).listen((event) async {
+      if (event == FGBGType.foreground && _shouldSync()) {
+        await _sync();
+      }
+    });
+
+    // Force a sync after Network is back
+    // TODO: Liquid SDK - This sync should happen on SDK layer after re-establishing connection
```

Comment:
Is this an issue intended for the SDK?

If so, then I'm not sure it's doable, because we don't have connectivity monitoring there.